core/src/main/java/com/opensymphony/xwork2/config/impl/DefaultConfiguration.java

```diff
@@ -459,9 +460,12 @@ protected synchronized RuntimeConfiguration buildRuntimeConfiguration() throws C
         boolean appendNamedParameters = Boolean.parseBoolean(
                 container.getInstance(String.class, StrutsConstants.STRUTS_MATCHER_APPEND_NAMED_PARAMETERS)
         );
+        boolean fallbackToEmptyNamespace = Boolean.parseBoolean(
+                Optional.ofNullable(container.getInstance(String.class, StrutsConstants.STRUTS_ACTION_CONFIG_FALLBACK_TO_EMPTY_NAMESPACE)).orElse("true")
```

Comment:
Instead of this fallback I would create a new entry into [default.properties](https://github.com/apache/struts/blob/master/core/src/main/resources/org/apache/struts2/default.properties) with value set to `true` and document it - this will help others understand the change

Reply:
@jefferyxhy and I just discussed this one and one of the drawbacks of putting it in `default.properties` is that it isn't read by unit tests and causes a bunch of test failures, as the unit tests will default to `false`. To get around this we could additionally add the constant to `StrutsDefaultConfigurationProvider` as well as `default.properties`.

I'm personally not too fussed. In the past I've deliberately made constants default to false to try sidestep this issue. Let us know know what you would prefer

Reply:
Understood, yet `default.properties` is used as a documentation by users so I would define the constant there with short description and keep fallback to `true` as above

Reply:
Updated. Thanks